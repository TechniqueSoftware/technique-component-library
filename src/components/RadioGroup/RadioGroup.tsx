import * as React from 'react';
import FormControlLabel from '@material-ui/core/FormControlLabel';
import FormLabel from '@material-ui/core/FormLabel';
import FormControl from '@material-ui/core/FormControl';
import MuiRadioGroup from '@material-ui/core/RadioGroup';
import Radio from '@material-ui/core/Radio';

interface RadioProps {
  value: string;
  label: React.ReactNode;
  disabled?: boolean;
}

export interface RadioGroupProps {
  onChange: (event: React.ChangeEvent<HTMLInputElement>, value: string) => void;
  label: string;
  name: string;
  value: string;
  radioPropsList: RadioProps[];
}

function renderRadioPropsList(formControlLabelPropsList: RadioProps[]) {
  return formControlLabelPropsList.map((props) => {
    return <FormControlLabel {...props} control={<Radio />} key={props.value} />;
  });
}

export function RadioGroup(props: RadioGroupProps) {
  const { onChange, label, name, value, radioPropsList } = props;
  return <FormControl component="fieldset">
    <FormLabel component="legend">{label}</FormLabel>
    <MuiRadioGroup aria-label={label} name={name} value={value} onChange={onChange}>
      {renderRadioPropsList(radioPropsList)}
    </MuiRadioGroup>
  </FormControl>;
}
