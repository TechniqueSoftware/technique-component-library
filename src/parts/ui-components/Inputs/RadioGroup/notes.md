# RadioGroup
This component wraps several components including the `Radio`, `RadioGroup`, `FormControl`, `FieldSet` from Material-UI
to create an easier abstraction. See the Material-UI [radio-buttons](https://material-ui.com/components/radio-buttons/)
docs for more advanced usages.

## JIRA
Related jira ticket: https://clubos.atlassian.net/browse/CR-714
