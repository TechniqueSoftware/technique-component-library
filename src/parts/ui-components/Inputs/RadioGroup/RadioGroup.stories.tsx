import * as React from 'react';
import ThemeProvider from '@material-ui/styles/ThemeProvider';
import { createCUICStoriesOf } from '../../../../components/UI-Components/utils/utils';
import { MODULE_NAME } from '../../../../components/UI-Components/constants';
import { text } from '@storybook/addon-knobs';
import { RadioGroupProps, RadioGroup } from '../../../../components/UI-Components/RadioGroup/RadioGroup';
import clubOsTheme from '../../../../components/UI-Components/storyThemes';
import Box from '@material-ui/core/Box';
import { IconTooltip, IconTooltipProps } from '../../../..';
import HelpRounded from '@material-ui/icons/HelpRounded';

function RadioGroupStory() {

  const [value, setValue] = React.useState('');
  const label = text('Label', 'Select an item');
  const iconTooltipProps: IconTooltipProps = {
    iconProps: {
      component: HelpRounded
    },
    tooltipText: 'Icons can be optionally added to radio buttons.'
  };
  const icon = <IconTooltip {...iconTooltipProps} />;

  const props: RadioGroupProps = {
    label,
    value,
    onChange: (event, value) => setValue(value),
    radioPropsList: [
      { label: 'Option 1', value: '1' },
      { label: 'Option 2', value: '2' },
      { label: <span>Option 3 {icon}</span>, value: '3' },
      { label: 'Option 4 (disabled)', value: '4', disabled: true }
    ],
    name: 'options'
  };

  return (
    <div>
      <ThemeProvider theme={clubOsTheme}>
        <Box p={2}>
          <RadioGroup {...props} />
        </Box>
      </ThemeProvider>
    </div>
  );
}

createCUICStoriesOf(MODULE_NAME.INPUTS, module)
  .addCUICStory({
    render: () => {
      return (<RadioGroupStory />);
    },
    name: 'RadioGroup',
    notes: require('./notes.md').default
  });
