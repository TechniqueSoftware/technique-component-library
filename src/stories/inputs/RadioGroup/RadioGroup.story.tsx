import * as React from 'react';
import ThemeProvider from '@material-ui/styles/ThemeProvider';
import { createCUICStoriesOf } from '../../utils/utils';
import { MODULE_NAME } from '../../constants';
import { text } from '@storybook/addon-knobs';
import { RadioGroupProps, RadioGroup } from '@clubos-ui-components/core/dist/components/RadioGroup/RadioGroup';
import clubOsTheme from '@clubos-ui-components/core/dist/themes/clubOS';
import Box from '@material-ui/core/Box';
import { IconTooltip, IconTooltipProps } from '@clubos-ui-components/core';
import HelpRounded from '@material-ui/icons/HelpRounded';

function RadioGroupStory() {

  const [value, setValue] = React.useState('');
  const label = text('Label', 'Select an item');
  const iconTooltipProps: IconTooltipProps = {
    iconProps: {
      component: HelpRounded
    },
    tooltipText: 'Icons can be optionally added to radio buttons.'
  };
  const icon = <IconTooltip {...iconTooltipProps} />;

  const props: RadioGroupProps = {
    label,
    value,
    onChange: (event, value) => setValue(value),
    radioPropsList: [
      { label: 'Option 1', value: '1' },
      { label: 'Option 2', value: '2' },
      { label: <span>Option 3 {icon}</span>, value: '3' },
      { label: 'Option 4 (disabled)', value: '4', disabled: true }
    ],
    name: 'options'
  };

  return (
    <div>
      <ThemeProvider theme={clubOsTheme}>
        <Box p={2}>
          <RadioGroup {...props} />
        </Box>
      </ThemeProvider>
    </div>
  );
}

createCUICStoriesOf(MODULE_NAME.INPUTS, module)
  .addCUICStory({
    render: () => {
      return (<RadioGroupStory />);
    },
    name: 'RadioGroup',
    notes: require('./notes.md').default
  });
