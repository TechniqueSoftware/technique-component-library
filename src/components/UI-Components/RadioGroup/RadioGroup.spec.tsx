import * as React from 'react';
import { RadioGroup, RadioGroupProps } from './RadioGroup';
import { shallow } from 'enzyme';

describe('RadioGroup', () => {
  test('should render with minimal props', () => {
    const props: RadioGroupProps = {
      radioPropsList: [
        { value: '1', label: 'foo' },
        { value: '2', label: 'bar' },
        { value: '3', label: 'zap', disabled: true }
      ],
      label: 'mockLabel',
      name: 'mockName',
      onChange: jest.fn(),
      value: 'mockValue'
    };
    const shallowWrapper = shallow(<RadioGroup {...props} />);
    expect(shallowWrapper.debug()).toMatchSnapshot();
  });
});
