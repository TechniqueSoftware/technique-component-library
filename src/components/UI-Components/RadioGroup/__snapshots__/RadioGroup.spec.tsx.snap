// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RadioGroup should render with minimal props 1`] = `
"<WithStyles(ForwardRef(FormControl)) component=\\"fieldset\\">
  <WithStyles(ForwardRef(FormLabel)) component=\\"legend\\">
    mockLabel
  </WithStyles(ForwardRef(FormLabel))>
  <ForwardRef(RadioGroup) aria-label=\\"mockLabel\\" name=\\"mockName\\" value=\\"mockValue\\" onChange={[Function: mockConstructor] { _isMockFunction: true, getMockImplementation: [Function (anonymous)], mock: Object [Object: null prototype] { calls: [], instances: [], invocationCallOrder: [], results: [] }, mockClear: [Function (anonymous)], mockReset: [Function (anonymous)], mockRestore: [Function (anonymous)], mockReturnValueOnce: [Function (anonymous)], mockResolvedValueOnce: [Function (anonymous)], mockRejectedValueOnce: [Function (anonymous)], mockReturnValue: [Function (anonymous)], mockResolvedValue: [Function (anonymous)], mockRejectedValue: [Function (anonymous)], mockImplementationOnce: [Function (anonymous)], mockImplementation: [Function (anonymous)], mockReturnThis: [Function (anonymous)], mockName: [Function (anonymous)], getMockName: [Function (anonymous)] }}>
    <WithStyles(ForwardRef(FormControlLabel)) value=\\"1\\" label=\\"foo\\" control={{...}} />
    <WithStyles(ForwardRef(FormControlLabel)) value=\\"2\\" label=\\"bar\\" control={{...}} />
    <WithStyles(ForwardRef(FormControlLabel)) value=\\"3\\" label=\\"zap\\" disabled={true} control={{...}} />
  </ForwardRef(RadioGroup)>
</WithStyles(ForwardRef(FormControl))>"
`;
